src/traces/scatter/index.js

```diff
@@ -34,7 +34,7 @@ Scatter.animatable = true;
 Scatter.moduleType = 'trace';
 Scatter.name = 'scatter';
 Scatter.basePlotModule = require('../../plots/cartesian');
-Scatter.categories = ['cartesian', 'symbols', 'markerColorscale', 'errorBarsOK', 'showLegend', 'scatter-like'];
+Scatter.categories = ['cartesian', 'svg', 'symbols', 'markerColorscale', 'errorBarsOK', 'showLegend', 'scatter-like', 'draggedPts'];
```

Comment:
**IMPORTANT** (and possibly worth refactoring)

scattergl and splom traces are currently declared with `'cartesian'` category so that cartesian axis logic that they share just works. But this means that not all 'cartesian' traces are _svg_ or more precisely not all 'cartesian' traces are plotted in `<g .cartersianlayer>`. So to avoid looping over all subplots on drag to shift `plotinfo.plot` and its clip `<rect>` for scattergl and splom traces, I made an `'svg'` category. This is a slight misnomer though, as heatmaps and histogram2d are here considered _svg_ even though they are built using a `<canvas>`. Oh well. The 'svg' category mirrors the 'gl' and 'regl' category listed in scattergl and splom trace modules. Any objections?

Moreover, I added a 'draggedPts' category to avoid those costly `selectAll('.point')` on pan when we don't need them. For example, graphs with contour lines feel faster on pan as we no longer traverse the `<g .cartesianlayer>` looking for points to scaled.

Reply:
I think this is fine - `'cartesian'` means "data relates to cartesian axes" and `'svg'` means "data is drawn into an SVG container" (heatmaps use a canvas to generate the image but then put it into an svg `<image>` element)

`'draggedPts'` is still potentially overkill, as the trace type doesn't necessarily tell you if the traces contain these elements - all of these trace types can be drawn with no points or text, only lines and fills...